### ISSUE TITLE
Holidays need duration

### ISSUE THREAD

**Issue by okhowang** [rickar/cal#105]
In some country, holiday may last several days, for example china spring festival, korea lunar new year.
in https://github.com/rickar/cal/pull/63, three holidays are defined. BeforeLunarNewYear/LunarNewYear/AfterLunarNewYear.
If holiday has duration, it can be more graceful

**Comment by rickar** [comment]
Agreed. It may require a special flag to mark extended holidays so the existing logic is kept simple.
